tests/app/Jobs/NotifyUserOfNewGistCommentTest.php

```diff
@@ -77,7 +77,8 @@ public function itGeneratesUnsubscribeUrl()
             'github_id' => 987654,
             'token' => 'ABC123',
         ]);
+        $expectedHash = '538240242f92164ed9c1bd413a094a1c0d70d5cca00d10478a077d42798aa2c6';
 
-        $this->assertEquals('https://giscus.co/unsubscribe?id=987654&hash=2e8f94ecbd8dba1da15c888b2ef0dbd3', $user->getUnsubscribeUrl());
+        $this->assertContains($expectedHash, $user->getUnsubscribeUrl());
```

Comment:
Looks good. I think I had asked for a test to cover that method that was wrongly named--my brain is only 10% here, but can we see if we can get a test that asserts more than just that the hash is calculated the way we're expecting? Feels like there should be some sort of more functional test we could write. Holler if you want to collaborate on figuring out what that would be.

Reply:
Yeah, I think we should test two things:
1. Is the link being constructed correctly for a given user?
2. When the link is used, is the end result that the user is shown the unsubscribe confirmation page?

The first can be accomplished through this method (maybe revising it to be a full URL comparison) but that'd have to be constructed using the same code in `App\User` and that seems *wrong* for some reason, since it'd basically be the exact same code found in `App\User@getUnsubscribeUrl`.

The second would be a kind of integration test, seeding a new user, generating a link, following the link, then asserting the expected view is there.

If you have different thoughts, let's discuss more, either email (jake.bathman [at] gmail)/Telegram/whatever's easiest. I'll have time later today to write a few more tests.

Reply:
@jakebathman I think the second one is definitely good. Given this link is visited, is the correct result that the user is unsubscribed?

Also, is unsubscribed already a concept in the app? If it's new we may also need to test that.

And yah, it's hard to test the URL generation without just duplicating the code. Not sure how to optimize that...